### PR TITLE
android: little build fix

### PR DESCRIPTION
**Pull request by devnexen**
Android support getentropy since the 28 version.

### CODE REVIEW

File: src/unix/android-ifaddrs.c [hunk 1]
```diff
@@ -35,6 +35,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <netinet/in.h>
 #include <linux/netlink.h>
 #include <linux/rtnetlink.h>
+#include <linux/if_packet.h>
```
Comment (saghul) [inline src/unix/android-ifaddrs.c]: Why is this needed?

File: src/unix/android-ifaddrs.c [hunk 2]
```diff
@@ -475,6 +476,7 @@ static int interpretAddr(struct nlmsghdr *p_hdr, struct ifaddrs **p_resultList,
                     l_addrSize += NLMSG_ALIGN(calcAddrLen(l_info->ifa_family, l_rtaDataSize));
                     l_addedNetmask = 1;
                 }
+                break;
```
Comment (saghul) [inline src/unix/android-ifaddrs.c]: What does this fix?

File: src/unix/android-ifaddrs.c (hunk 1 above)
Comment (devnexen) [reply src/unix/android-ifaddrs.c]: otherwise it does not build (sockaddr_ll definition).

File: src/unix/android-ifaddrs.c (hunk 2 above)
Comment (devnexen) [reply src/unix/android-ifaddrs.c]: unintended fall through case.

File: src/random.c [hunk 3]
```diff
@@ -35,7 +35,7 @@ static int uv__random(void* buf, size_t buflen) {
   rc = uv__random_readpath("/dev/urandom", buf, buflen);
 #elif defined(_AIX)
   rc = uv__random_readpath("/dev/random", buf, buflen);
-#elif defined(__APPLE__) || defined(__OpenBSD__)
+#elif defined(__APPLE__) || defined(__OpenBSD__) || (defined(__ANDROID_API__) && __ANDROID_API__ >= 28)
```
Comment (bnoordhuis) [inline src/random.c]: Long line, should be <= 80 columns.

You don't need to check the API version because libuv will fall back to /dev/urandom when `getentropy` isn't dlsym-able but I don't have strong feelings one way or the other.

**Comment by cjihrig** [comment]
> Can you split the commit in two? They're two unrelated changes mashed together now.

Ping @devnexen regarding this ^

**Comment by devnexen** [comment]
Sorry I missed that bit.

**Comment by saghul** [comment]
Landed on f261d04d0a36df7c8ab041a45399c1fcba0a3c04 e83dba7f936b541bd37ac549d89502ec466c5ccf cheers!
